### DISCUSSION TITLE
Exceeded 100 jobs limit

### DISCUSSION THREAD

**Discussion by randybarton24** [jamessimone/apex-rollup#487] [Q&A]
Hi @jamessimone, still struggling to get the package working in our org.  We have a managed package that does a lot of batch processing overnight for our billing and subscriptions, which will trigger several rollups and probably cascading rollups.  It seems like that might be causing issues, as I've been getting failures happening almost every night around that time.  The most common being the "You've exceeded the limit of 100 jobs in the flex queue", followed by Apex heap size errors.  Do you have any recommendations?

![image](https://github.com/jamessimone/apex-rollup/assets/8461377/c80842f3-0e31-4149-a7a5-0e889d62139a)

**Comment by jamessimone** [comment]
@randybarton24 do you have any idea as to how many records are being updated by that managed package? I'm going to do some analysis on cascading rollups to see where this heap size issue is coming from. At the very least, I think you can get rid of the Flex Queue errors by reducing the Max Rollup Retries field on your rollup control CMDT record to 0.

**Reply by randybarton24** [reply]
@jamessimone ok I'll try that.  If it's the process that I'm thinking it is, it isn't that many records, but there are quite a few cascading records.  The parent to child object model is Account>Opportunity>Product Subscription>Billing Schedule... so it's a renewal process that would be operating on a few hundred Accounts each night.  It would create a new Opportunity for each of those Accounts, and upon closing the Opportunity it would generate maybe 2-7 Product Subscriptions per Opportunity, and each of those Product Subscriptions would have 12 Billing Schedules, one for each monthly period in the upcoming year.  We have rollups configured for each of those parent/child relationships.

**Reply by jamessimone** [reply]
@randybarton24 with the latest update, you should be able to re-up that Max Rollup Retries number and stop running into the heap size limits (which were then cascading into a set of other errors, I believe).

**Reply by randybarton24** [reply]
@jamessimone ok thank you, I'll give it a shot!
